src/store.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { useCallback, useEffect, useState } from "react";
+import { Status } from "./types/Status";
+import { TodosContextType } from "./types/TodosContextType";
+import { useLocalStorage } from "./Hooks/useLocalStorage";
+import { Todo } from "./types/Todo";
+
+const initialTodos: TodosContextType = {
+  todos: [],
+  filteredTodos: [],
+  addTodo: () => {},
+  removeTodo: () => {},
+  editTodo: () => {},
+  setTodos: () => {},
+  status: Status.All,
+  setStatus: () => {},
+};
+
+export const TodosContext = React.createContext(initialTodos);
+
+interface Props {
+  children: React.ReactNode;
+}
+
+export const TodosContextProvider: React.FC<Props> = ({ children }) => {
+  const [todos, setTodos] = useLocalStorage<Todo[]>("todos", []);
+  const [status, setStatus] = useState(Status.All);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
```

Comment:
Do not use extra state + useEffect. Create function helper that will return visible todos
```suggestion
  const filteredTodos = getVisibleTodos(todos, status)
```